admin/class-yoast-network-admin.php

```diff
@@ -0,0 +1,248 @@
+		wp_safe_redirect( $sendback );
+		exit;
+	}
+
+	/**
+	 * Outputs nonce, action and option group fields for a network settings page in the plugin.
+	 *
+	 * @param string $option_group Option group name for the current page.
+	 *
+	 * @return void
+	 */
+	public function settings_fields( $option_group ) {
+		?>
+		<input type="hidden" name="network_option_group" value="<?php echo esc_attr( $option_group ); ?>" />
+		<input type="hidden" name="action" value="<?php echo esc_attr( self::UPDATE_OPTIONS_ACTION ); ?>" />
+		<?php
+		wp_nonce_field( "$option_group-network-options" );
+	}
+
+	/**
+	 * Prints the form for restoring a site with its default settings.
+	 *
+	 * @return void
+	 */
+	public function print_restore_form() {
+		$yform = Yoast_Form::get_instance();
+
+		echo '<h2>' . esc_html__( 'Restore site to default settings', 'wordpress-seo' ) . '</h2>';
+		echo '<form action="' . esc_url( network_admin_url( 'admin.php' ) ) . '" method="post" accept-charset="' . esc_attr( get_bloginfo( 'charset' ) ) . '">';
+		wp_nonce_field( 'wpseo-network-restore' );
+		echo '<p>' . esc_html__( 'Using this form you can reset a site to the default SEO settings.', 'wordpress-seo' ) . '</p>';
+
+		if ( get_blog_count() <= 100 ) {
+			$yform->select(
+				'site_id',
+				__( 'Site ID', 'wordpress-seo' ),
+				$this->get_site_choices()
+			);
+		}
+		else {
+			$yform->textinput( 'site_id', __( 'Blog ID', 'wordpress-seo' ) );
+		}
+
+		echo '<input type="hidden" name="action" value="' . esc_attr( self::RESTORE_SITE_ACTION ) . '" />';
+		echo '<input type="submit" name="wpseo_restore_blog" value="' . esc_attr__( 'Restore site to defaults', 'wordpress-seo' ) . '" class="button"/>';
+		echo '</form>';
+	}
+
+	/**
+	 * Hooks in the necessary actions and filters.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+
+		if ( $this->hooks_registered ) {
+			return;
+		}
+
+		$this->hooks_registered = true;
+
+		add_action( 'admin_action_' . self::UPDATE_OPTIONS_ACTION, array( $this, 'handle_update_options_request' ) );
+		add_action( 'admin_action_' . self::RESTORE_SITE_ACTION, array( $this, 'handle_restore_site_request' ) );
+		add_action( 'wpseo_admin_footer', array( $this, 'print_restore_form' ) );
+	}
+
+	/**
+	 * Persists settings errors.
+	 *
+	 * Settings errors are stored in a transient for 30 seconds so that this transient
+	 * can be retrieved on the next page load.
+	 *
+	 * @return void
+	 */
+	private function persist_settings_errors() {
+
+		// Use a regular transient here, since it is automatically cleared right after the redirect.
+		// A network transient would be cleaner, but would require a lot of copied code from core for
+		// just a minor adjustment when displaying settings errors.
+		set_transient( 'settings_errors', get_settings_errors(), 30 );
+	}
+
+	/**
+	 * Gets the singleton instance of this class.
+	 *
+	 * @return Yoast_Network_Admin The singleton instance.
+	 */
+	public static function get() {
```

Comment:
I am not a big fan of singleton. It got us in situations where unexpected things happened.

Reply:
See my previous comment.